#!/usr/bin/env python
PACKAGE = "stereo_camera_pose_estimation"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("maximum_depth", double_t, 0, "An Integer parameter", 2.00, 0.0, 15.0)
gen.add("aruco_id", int_t, 0, "An Integer parameter", 0, 0, 50)
gen.add("aruco_size", double_t, 0, "An Integer parameter", 0.02, 0.0, 0.03)
gen.add("max_yaw", double_t, 0, "An Integer parameter", 0.785, 0.0, 3.141)
gen.add("max_pitch", double_t, 0, "An Integer parameter", 0.785, 0.0, 3.141)
gen.add("max_roll", double_t, 0, "An Integer parameter", 0.785, 0.0, 3.141)
gen.add("idle_time", double_t, 0, "An Integer parameter", 10.0, 0.0, 60.0)
gen.add("x_offset", double_t, 0, "An Integer parameter", 0.0, -1.0, 1.0)
gen.add("y_offset", double_t, 0, "An Integer parameter", 0.0, -1.0, 1.0)
gen.add("height_offset", double_t, 0, "An Integer parameter", 0.0, -1.0, 1.0)
gen.add("yaw_offset", double_t, 0, "An Integer parameter", 0.0, -1.0, 1.0)
gen.add("pitch_offset", double_t, 0, "An Integer parameter", 0.0, -1.0, 1.0)
gen.add("roll_offset", double_t, 0, "An Integer parameter", 0.0, -1.0, 1.0)
gen.add("debug", bool_t, 0, "An Integer parameter", True)

exit(gen.generate(PACKAGE, "stereo_camera_pose_estimation", "StereoCameraPoseEstimation"))
